File: bindings/java/src/main/java/io/omq/SocketType.java
package io.omq;

/** OMQ socket type. */
public enum SocketType {
    /** Strict request socket. */
    REQ(0),
    /** Strict reply socket. */
    REP(1),
    /** Publish socket. */
    PUB(2),
    /** Subscribe socket. */
    SUB(3),
    /** Extended publish socket. */
    XPUB(4),
    /** Extended subscribe socket. */
    XSUB(5),
    /** Pipeline push socket. */
    PUSH(6),
    /** Pipeline pull socket. */
    PULL(7),
    /** Async request socket. */
    DEALER(8),
    /** Async reply routing socket. */
    ROUTER(9),
    /** One-to-one bidirectional socket. */
    PAIR(10),
    /** Client socket. */
    CLIENT(11),
    /** Server socket. */
    SERVER(12),
    /** Radio group publish socket. */
    RADIO(13),
    /** Dish group subscribe socket. */
    DISH(14),
    /** Single-part pipeline push socket. */
    SCATTER(15),
    /** Single-part pipeline pull socket. */
    GATHER(16),
    /** Single-part one-to-one socket. */
    CHANNEL(17),
    /** Peer socket. */
    PEER(18),
    /** Raw TCP stream socket. */
    STREAM(19);

    private final int code;

    SocketType(int code) {
        this.code = code;
    }

    int code() {
        return code;
    }
}

File: bindings/dotnet/src/SocketType.cs
namespace Omq;

/// OMQ socket patterns supported by the native ABI.
public enum SocketType
{
    Pair = 0, Pub = 1, Sub = 2, Req = 3, Rep = 4, Dealer = 5,
    Router = 6, Pull = 7, Push = 8, XPub = 9, XSub = 10, Stream = 11,
    Server = 12, Client = 13, Radio = 14, Dish = 15, Gather = 16,
    Scatter = 17, Dgram = 18, Peer = 19, Channel = 20
}
